src/suricata.c

```diff
@@ -2698,6 +2692,12 @@ int main(int argc, char **argv)
         FlowDisableFlowRecyclerThread();
     }
 
+    if (suri.run_mode != RUNMODE_UNIX_SOCKET) {
+        /* we need a packet pool for FlowForceReassembly */
+        PacketPoolInit();
+        FlowForceReassembly();
```

Comment:
I don't know enough to answer that.

What I saw, was that the FlowForceReassembly() was called while processing flows, and after breaking main's loop , and as a side effect, the http output was borked.

For instance, by adding a single time(1) right after suri's main loop, the http output was complete and ok for the supplied pcap files in redmine ticket 1946.